Sources/SourceKit/sourcekitd/SwiftLanguageServer.swift

```diff
@@ -384,13 +384,42 @@ extension SwiftLanguageServer {
           }
           let isInsertTextSnippet = clientSupportsSnippets && text != insertText
 
+          let textEdit: TextEdit?
+          if let text = text {
+            let utf8BytesToErase: Int = value[self.keys.num_bytes_to_erase] ?? 0
+
+            var textEditRangeStart = completionPos
+            if utf8BytesToErase != 0 {
+              if let completionPosOffset = snapshot.lineTable.utf8OffsetOf(line: completionPos.line, utf16Column: completionPos.utf16index),
+                let textEditRangeStartLoc = snapshot.lineTable.lineAndUTF16ColumnOf(utf8Offset: completionPosOffset - utf8BytesToErase) {
+                textEditRangeStart = Position(line: textEditRangeStartLoc.line, utf16index: textEditRangeStartLoc.utf16Column)
```

Comment:
The spec says this needs to be on the same line:
> *Note:* The range of the edit must be a single line range and it must contain the position at which completion has been requested.

It's probably safest to just ignore `num_bytes_to_erase` when it crosses a line boundary.  You'll end up with something invalid

```
func
override func blah() {}
```

but I think it's understandable to users that we're only fixing it up on one line.  I doubt it's worth adding an extension just for this case.